ynr/apps/candidates/models/popolo_extra.py

```diff
@@ -289,6 +301,43 @@ def has_results(self):
             return True
         return False
 
+    @property
+    def uncontested(self):
+        if not self.candidates_locked:
+            return False
+        if self.get_winner_count == self.membership_set.count():
+            return True
+        return False
+
+    def mark_uncontested_winners(self, request, ip_address, log=True):
+        """
+        If the election is uncontested mark all candidates as elected
+        """
+        if not self.uncontested:
+            return
+
+        self.membership_set.update(elected=True)
+
+        if not log:
+            return
+        LoggedAction.objects.create(
+            user=request.user,
+            action_type="constituency-{}".format("-lock"),
+            ip_address=ip_address,
+            ballot=self,
+            source="Ballot locked; Election is uncontested",
+        )
+
+    def unlock_ballot(self, request, ip_address, log=False):
+        self.membership_set.update(elected=False)
```

Comment:
We don't want to unset elected every time we unlock any ballot, I don't think. It would be better to only mark the candidates as not elected if the ballot is uncontested.

Also, I think `elected` is a `NullBooleanField`, e.g where `None` means "we don't know yet", `True` means "yes" and `False` means "no". Because of that, I think we should set this to `None` rather than `False` (but only for uncontested elections)